### DISCUSSION TITLE
Some requests get intercepted and others fail with *The client does not trust the proxy's certificate*

### DISCUSSION THREAD

**Discussion by mehov** [mitmproxy/mitmproxy#6847] [Q&A]
Hi everyone

I created a Virtual Device with old (pre-certificate pinning) Android and installed the MITM Proxy certificate on it.

The traffic gets intercepted inconsistently. 

When I look at the MITM Proxy console output (not browser), some requests do seem to work, while others fail with *The client does not trust the proxy's certificate*. The browser UI does, however, show some requests and I can click around a view the details/response for each of them.

Apps work inconsistently too.

- On one hand, I am able to browse inside the Play Store app. I made a search for something and I have seen the HTTPs requests show up in MITM Proxy browser UI.
- On the other hand, Google Maps says it can't connect to Internet. HTTPs websites in Chrome do not work either.

If it helps, here's a MITM Proxy console screenshot from when the device was started, but still locked and not interacted with.

![1](https://github.com/mitmproxy/mitmproxy/assets/7813306/59d32146-6e9d-4ff1-a760-06b82761420f)

I'm on a virtual device using Android 7.0 x86 API 24.

Is this familiar or a known issue? I have made sure I uploaded the right certificate from the right user folder to the device. But then again, if I uploaded a wrong certificate, then I wouldn't be seeing any connections?

**Comment by mehov** [comment]
After reading https://github.com/mitmproxy/mitmproxy/discussions/6621, I think I should clarify that I only configured the AVD proxy in the LTE settings (so not even Wi-Fi, because the AVD doesn't show `AndroidWifi` for some reason).

**Comment by Prinzhorn** [comment]
> old (pre-certificate pinning) Android
> I'm on a virtual device using Android 7.0 x86 API 24.

Could you clarify what you consider "pre-certificate pinning"?

**Comment by mehov** [comment]
> > old (pre-certificate pinning) Android
> > I'm on a virtual device using Android 7.0 x86 API 24.
> 
> Could you clarify what you consider "pre-certificate pinning"?

I thought Android 7.0 x86 API 24 was pre-pinning, but I was probably wrong. Should I install Android 6 or earlier?

**Reply by Prinzhorn** [reply]
I'm no Android expert, but what would keep an app in Android 1 from manually implementing certificate pinning? I'm not surprised official Google apps (Google Maps, Chrome) are pinning Google services in Android 7 (and likely below).

I'd look into tools to remove the pinning, see https://docs.mitmproxy.org/stable/concepts-certificates/#certificate-pinning
